node/pkg/watchers/aptos/watcher.go

```diff
@@ -196,35 +223,47 @@ func (e *Watcher) Run(ctx context.Context) error {
 				logger.Error("InvalidJson: " + string(eventsJson))
 				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
 				continue
-
 			}
 
-			events := gjson.ParseBytes(eventsJson)
+			// Parse the GraphQL response
+			response := gjson.ParseBytes(eventsJson)
 
-			// the endpoint returns an array of events, ordered by sequence
-			// id (ASC)
-			for _, event := range events.Array() {
-				eventSequence := event.Get("sequence_number")
-				if !eventSequence.Exists() {
+			// Extract the msg array from data.msg
+			messages := response.Get("data.msg")
+			if !messages.Exists() {
+				logger.Error("No data.msg field in indexer response")
+				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
+				continue
+			}
+
+			// Walk through the array of events
+			for _, msg := range messages.Array() {
+				version := msg.Get("version")
+				sequenceNum := msg.Get("sequence_num")
+				if !version.Exists() || !sequenceNum.Exists() {
 					continue
 				}
 
-				eventSeq := eventSequence.Uint()
-				if nextSequence == 0 && eventSeq != 0 {
```

Comment:
should this block stay? what prevents this new version from publishing the last event every startup?